qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -761,21 +761,36 @@ def delete(cls, id_):
         conn_handler = SQLConnectionHandler()
 
         # Delete the sample template filepaths
-        conn_handler.execute(
-            "DELETE FROM qiita.sample_template_filepath WHERE "
-            "study_id = %s", (id_, ))
+        queue = "delete_sample_template_%d" % id_
+        conn_handler.create_queue(queue)
 
-        conn_handler.execute(
-            "DROP TABLE qiita.{0}".format(table_name))
-        conn_handler.execute(
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.sample_template_filepath WHERE study_id = %s",
+            (id_, ))
+
+        conn_handler.add_to_queue(
+            queue,
+            "DROP TABLE IF EXISTS qiita.{0}".format(table_name))
+
+        conn_handler.add_to_queue(
```

Comment:
Drop the table twice? Also it should probably just be `drop table`, that way we get an error if the table doesn't exist. This will point out that the DB is in an inconsistent state.